### ISSUE TITLE
`match_overlapping_arm` false positive when ranges have negative starts

### ISSUE THREAD

**Issue by mikerite** [rust-lang/rust-clippy#7829]
Lint name: match_overlapping_arm

I tried this code:

```rust
#[warn(clippy::match_overlapping_arm)]

fn main() {
    match 0 {
        -1..=1 => (),
        -2..=2 => (),
        _ => (),
    }
}
```

I expected to see this happen: No warning. Although it's not documented the lint should not warn when a range is 'completely included' in another one.

Instead, this happened: 
```
warning: some ranges overlap
 --> src/main.rs:5:9
  |
5 |         -1..=1 => (),
  |         ^^^^^^
  |
note: the lint level is defined here
 --> src/main.rs:1:8
  |
1 | #[warn(clippy::match_overlapping_arm)]
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: overlaps with this
 --> src/main.rs:6:9
  |
6 |         -2..=2 => (),
  |         ^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_overlapping_arm
```

### Meta
Clippy 0.1.57 (2021-10-15 c102653) on the playground

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=1731d3e2cb3e8df9340cb6d3987dd986)
